src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.SearchResult.cs

```diff
@@ -59,13 +60,32 @@ private class SearchResult : INavigateToSearchResult
                 });
             }
 
-            private static string ConstructSecondarySortString(DeclaredSymbolInfo declaredSymbolInfo)
+            private static readonly char[] s_dotArray = { '.' };
+
+            private static string ConstructSecondarySortString(
+                Document document,
+                DeclaredSymbolInfo declaredSymbolInfo)
             {
-                var secondarySortString = string.Concat(
-                    declaredSymbolInfo.ParameterCount.ToString("X4"),
-                    declaredSymbolInfo.TypeParameterCount.ToString("X4"),
-                    declaredSymbolInfo.Name);
-                return secondarySortString;
+                var parts = ArrayBuilder<string>.GetInstance();
+                try
+                {
+                    parts.Add(declaredSymbolInfo.ParameterCount.ToString("X4"));
```

Comment:
You could leave these out of the array, and just have the last line be a string.Concat with these and the string.Join.  

There is no real point in putting these into the array every time.